nixos/modules/config/swap.nix

```diff
@@ -247,7 +275,12 @@ in
                 ''}
                 ${optionalString sw.randomEncryption.enable ''
                   cryptsetup plainOpen -c ${sw.randomEncryption.cipher} -d ${sw.randomEncryption.source} \
-                    ${optionalString sw.randomEncryption.allowDiscards "--allow-discards"} ${sw.device} ${sw.deviceName}
+                '' + concatLines (filter (s: s != "") [
```

Comment:
you can get rid of the filter by just using `optional` rather than `optionalString` and perform `concatStringsSep "\n"` or similar I believe.

Reply:
Managed to remove the `filter`, although `concatStringsSep` itself doesn't directly enjoy consuming lists so that trades the `filter` for a `flatten` which at least is a little simpler. It turned out that `concatMapStrings` was the right way to go after that since the continuing input mark needs to come after the last argument as well as between them.